docs/4.3. Behaviour - Tokens.md

```diff
@@ -141,38 +146,38 @@ claims, and ensure that enough space is available in the header for both the tok
   "aud": "https://node-*.example.com",
   "iat": "1548779460",
   "exp": "1548783060",
+  "scope": ["register", "query", "connection"],
+  "client_id": "hopy0dNRPNTiGJDqPfqYwGmw",
   "x-nmos-api": {
-    "register": {
-      "read": ["node.*", "query.*"],
-      "create": ["query.subscriptions", "registration.*"]
```

Comment:
I had forgotten that we had looked into using URL path fragments in the claims here, and I can't remember if we ruled it out for any reason. I wonder if there is still any value in it, allowing us to restrict implementations to (for example) use IS-05 single but not bulk, or receivers but not senders. If this were the case we'd use the following by default:
```
"connection": {
  "read": ["*"],
  "write": ["*"]
}
```

but could also do...

```
"connection": {
  "read": ["*"],
  "write": ["single/receivers/*", "bulk/receivers/*"]
}
```

We wouldn't want to permit IDs in this to avoid token bloat, but you could use a * in the middle of a path to wildcard our the individual resource IDs but restrict access to a sub-path.

I suspect we may have discounted it given that the approvals process and how it appears in a user interface may then be more complex for end users, indicating this isn't necessarily the best use of OAuth, but I'm unsure.

Reply:
I think we vaguely came to the conclusion that allowing path specifiers was a nice addition, but something that may not often be used, and harder for vendors to implement. Think we said something about only allowing wildcards at the end of paths?

But I agree that it would be useful. Now we've reduced the fields from CRUD to read/write, perhaps its worth looking at.